rfc/text/advanced/ap_10_rpc_progressive_call_invocations.md

```diff
@@ -234,9 +234,53 @@ The sequence diagram below illustrates this sitation, where the Network actor mo
         | <-----------|             |             |
         |             |             |             |
      ,--+---.     ,---+---.      ,--+---.    ,--+---.
-     |Caller|     |Network|      |Dealer|    |Dealer|
+     |Caller|     |Network|      |Dealer|    |Callee|
      `------'     `-------'      `------'    `------'
 
+From the *Callee*'s perspective, a call with progressive invocations is considered completed when the *Callee* sends a correlated final YIELD or ERROR message.
+
+Due to network delays, the *Dealer* may be unaware that the call is completed by time it sends another progressive INVOCATION.
+
+When a *Callee* receives an INVOCATION under the following conditions:
+
+- the *Callee* supports Progressive Call Invocations,
+- the INVOCATION request ID does not correspond to a new call request, and,
+- the INVOCATION request ID does not match any RPC invocation in progress,
+
+then it MUST ignore and discard that INVOCATION message without any further correlated response to the *Dealer*. The rationale for this is that the *Caller* will eventually receive a correlated RESULT or ERROR message from the previous call completion and will be able to handle the call completion accordingly.
```

Comment:
*Caller* should be *dealer* and RESULT should be YIELD here right?

Reply:
Caller should be dealer — right! And fixed! 
But RESULT should be RESULT as it is about Caller side in the end and not about Dealer side. So I think it is correct.

Reply:
But if you changed caller into dealer, but not RESULT into YIELD, then now it reads
> ... the *dealer* will eventually receive a correlated RESULT ...

And that's not correct right?

I think in this part the rationale is supposed to be about the dealer who doesn't need an extra ERROR for the raced INVOCATION since it will eventually always receive the ERROR/YIELD that has been delayed by the network...

Reply:
Sorry @Jopie64, it seems that I said it incorrectly. I reread it again. The final merged version:
```
The rationale for this is that the *Caller* will eventually receive a correlated RESULT or ERROR message from the previous call completion and will be able to handle the call completion accordingly.
```

Looks to be clear and correct. Or do you think that is not?

Reply:
Not sure... Also not sure how I could see the final version on my mobile 😅

There are two places with this rationale. One is about race between caller-dealer, and one between dealer-callee.

The one you quoted would be correct for the one between caller-dealer, but I thought this part was about the race between dealer-callee...

Reply:
Yeah. But in any case, the call ends on the Caller's side with the RESULT or ERROR. And that is stated. If I correctly understand you, you want to extend the description for the dealer-callee race with smth like that:

> The rationale for this is that the Dealer will eventually receive a correlated YIELD or ERROR message from the Callee and then send a correlated RESULT or ERROR message to the Caller so it will be able to handle the call completion accordingly.

Right?

Reply:
> Yeah. But in any case, the call ends on the Caller's side with the RESULT or ERROR. And that is stated. If I correctly understand you, you want to extend the description for the dealer-callee race with smth like that:
> 
> > The rationale for this is that the Dealer will eventually receive a correlated YIELD or ERROR message from the Callee and then send a correlated RESULT or ERROR message to the Caller so it will be able to handle the call completion accordingly.
> 
> Right?

I prefer this version of the paragraph. Shall I submit another PR with this change?